File: source/GyeongHo/boj/15650.cpp
#include <iostream>
using namespace std;

int arr[8];
int visited[9];
int n, m;

void pick(int cnt, int prev)
{
	if (cnt >= m) {
		for (int i = 0; i < m; ++i)
			cout << arr[i] << ' ';
		cout << '\n';
		return;
	}
	for (int i = prev; i <= n; ++i) {
		if (visited[i])
			continue;
		arr[cnt] = i;
		visited[i] = 1;
		pick(cnt + 1, i);
		visited[i] = 0;
	}
}

int main()
{
	ios::sync_with_stdio(0); cin.tie(0);
	cin >> n >> m;
	pick(0, 1);
	return 0;
}

File: source/GyeongHo/boj/15650.py
import sys

def pickM(arr, cnt, prev, n, m):
    if cnt >= m:
        for i in range(m):
            print(arr[i], end=' ')
        print()
        return
    for i in range(prev, n + 1):
        if visited[i]:
            continue
        arr[cnt] = i
        visited[i] = 1
        pickM(arr, cnt + 1, i, n, m)
        visited[i] = 0

n, m = map(int, sys.stdin.readline().split())
arr = [0] * n
visited = [0] * (n + 1)
pickM(arr, 0, 1, n, m)